nifi/nifi-nar-bundles/nifi-kite-bundle/nifi-kite-processors/src/main/java/org/apache/nifi/processors/kite/JSONFileReader.java

```diff
@@ -0,0 +1,111 @@
+/*
+ * Copyright 2015 Cloudera Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.nifi.processors.kite;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.google.common.base.Function;
+import com.google.common.base.Preconditions;
+import com.google.common.collect.Iterators;
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.Iterator;
+import javax.annotation.Nullable;
+import org.apache.avro.Schema;
+import org.apache.avro.generic.GenericData;
+import org.kitesdk.data.DatasetIOException;
+import org.kitesdk.data.spi.AbstractDatasetReader;
+import org.kitesdk.data.spi.DataModelUtil;
+import org.kitesdk.data.spi.JsonUtil;
+import org.kitesdk.data.spi.ReaderWriterState;
+
+/**
+ * This is a temporary addition. The version in 0.18.0 throws a NPE when the
+ * InputStream constructor is used.
+ */
+public class JSONFileReader<E> extends AbstractDatasetReader<E> {
```

Comment:
This one is in the NiFi namespace because it relies on public parts of the Kite SPI, it's the CSVFileReader that had to be in the Kite namespace.

The version of Kite is bundled in the NAR, so I think dropping in a different version is a low risk.

This should definitely be package-private, but we can't make it much smaller. The parent maintains its state and requires a call to initialize. We can't change either the `state` or `incoming` stream, so there isn't much to do but copy the class and fix this in the next version of Kite (due out very soon).